app.py

```diff
@@ -28,96 +30,122 @@ def init_db():
 
 ###CREATION AND DELETION FUNCTIONS
 
-#Function to create a note
-def create_note(): 
-    pass 
-
-#function to delete a note
-def delete_note():
-    pass
-
-#function to create a tag
-def create_tag(): 
-    pass
-
-#function to delete a tag
-def delete_tag():
-    pass 
-
-#function to add content
-#I do not know if we need a separate function for edit content. I can explain the reasoning in class 
-def add_content(): 
-    pass 
-
-
-###SEARCHING FUNCTIONS
-
-#This will be called by the frontend for two different functions
-    #1) Viewing Content 
-    #2) PDF Conversion 
-def return_content():
-    pass 
-
-#function to search by title 
-def search_by_title():
-    pass 
-
-#function to search by modified date
-def search_by_modified_date():
-    pass 
-
-#function to search by created date
-def search_by_created_date():
-    pass
-
-#function to search by tag
-def search_by_tag(): 
-    pass
-
-
-
-###LISTING FUNCTIONS
+#Used to create notes and create tags
+@app.route('/create', methods=['POST'])
+def create(): 
+    conn = sqlite3.connect("note.db")
+    cursor = conn.cursor()
+    payload = request.json
+    try: 
+        title = payload.get('title')
+    except KeyError:
+        abort(206, "Data Entry Requires A Title")
+    
+    try: 
+        content = payload.get('content')
+    except KeyError: 
+        content = ""
+    
+    try: 
+        tags = payload.get('tag')
+    except KeyError: 
+        tags = ""
+    
+    if tags != "":
+        for value in tags: 
+            cursor.execute("INSERT INTO tags VALUES (?,?)", (title, value))
+    
+    else: 
```

Comment:
this shouldn't be an else statement! just an if is required.

a proof of correctness could look like this: imagine for a second that we passed some tags along with this call. tags wouldn't be empty and so we would execute the "if" block. but then the "else" block would be skipped. we wouldn't want that